src/main/java/com/textadventure/Lexer.java

```diff
@@ -26,8 +25,11 @@ private TokenType(String pattern) {
 			tokenPatternsBuffer.append(String.format("|(?<%s>%s)", tokenType.name(), tokenType.pattern));
 		Pattern tokenPatterns = Pattern.compile(new String(tokenPatternsBuffer.substring(1)));
```

Comment:
`Pattern.compile(new String(tokenPatternsBuffer.substring(1)));`
Here new String() is redundant. `tokenPatternsBuffer.substring(1)` will give you String only.